### PR TITLE
Return average score for a restaurant along with all of its reviews

### PR DESCRIPTION
**Pull request by NickPriv**
- Create object `GetAllReviewsOutput` which holds both the reviews and the average score
- Create aggregation which filters for a restaurantID and computes the average score
- Updating and adding unit tests accordingly

This branch won't be merged into mainline until the frontend is set up to handle the new output from the reviews endpoint

### CODE REVIEW

File: src/main/java/com/fryrank/dal/ReviewDALImpl.java [hunk 1]
```diff
@@ -1,26 +1,55 @@
...
+
+        final MatchOperation filterToRestaurantId = match(equalToRestaurantIdCriteria);
+        final GroupOperation averageScore = group(RESTAURANT_ID_KEY).avg("score").as("avgScore");
+        final Aggregation aggregation = newAggregation(filterToRestaurantId, averageScore);
+        final AggregationResults<RestaurantAvgScore> result = mongoTemplate.aggregate(aggregation, "review", RestaurantAvgScore.class);
+        final Optional<RestaurantAvgScore> uniqueResult = Optional.ofNullable(result.getUniqueMappedResult());
+
+        if (uniqueResult.isPresent()) {
+            final Float rawScore = uniqueResult.get().getAvgScore();
+            final BigDecimal averageScoreBigDecimal = new BigDecimal(rawScore).setScale(2, RoundingMode.DOWN);
```
Comment (oxyflush) [inline src/main/java/com/fryrank/dal/ReviewDALImpl.java:47]:
```suggestion
            final BigDecimal averageScoreBigDecimal = new BigDecimal(rawScore).setScale(1, RoundingMode.DOWN);
```

File: src/test/java/com/fryrank/dal/ReviewDALTests.java [hunk 2]
```diff
@@ -24,21 +31,44 @@ public class ReviewDALTests {
...
+        when(mongoTemplate.find(query, Review.class)).thenReturn(expectedReviews);
+
+        final AggregationResults<RestaurantAvgScore> aggregationResults = new AggregationResults<>(List.of(), new Document());
+        when(mongoTemplate.aggregate(any(Aggregation.class), anyString(), eq(RestaurantAvgScore.class))).thenReturn(aggregationResults);
 
-        assertEquals(actualReviews, expectedReviews);
+        final GetAllReviewsOutput expectedOutput = new GetAllReviewsOutput(expectedReviews, null);
+        final GetAllReviewsOutput actualOutput = reviewDAL.getAllReviewsByRestaurantId(TEST_RESTAURANT_ID);
+        assertEquals(actualOutput, expectedOutput);
     }
```
Comment (oxyflush) [inline src/test/java/com/fryrank/dal/ReviewDALTests.java:72]: Test for `getAllReviewsByRestaurantId()` with null restaurant ID

**Review by oxyflush — COMMENTED** [review]
Just need to add test for null restaurant ID https://github.com/NickPriv/FryRankBackend/pull/49/files#r1507064750

**Review by oxyflush — APPROVED** [review]

File: src/main/java/com/fryrank/dal/ReviewDALImpl.java (hunk 1 above)
Comment (NickPriv) [reply src/main/java/com/fryrank/dal/ReviewDALImpl.java]: Making this change

File: src/test/java/com/fryrank/dal/ReviewDALTests.java (hunk 2 above)
Comment (NickPriv) [reply src/test/java/com/fryrank/dal/ReviewDALTests.java:72]: Sure, adding it. Expected result would be NullPointerException
